File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode curr = head;
        ListNode pre = null;
        ListNode Next = null;
        while(curr!=null){
            Next = curr.next;
            curr.next = pre;
            pre = curr;
            curr = Next;
        }
        return pre;
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
from collections import deque
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution(object):
    def reverseList(self, head):
        self.head = head
        curr = self.head
        stack = deque()
        while curr is not None:
            stack.append(curr.val)
            curr = curr.next
        
        curr = self.head
        while len(stack)>0:
            ele = stack.pop()
            curr.val = ele
            curr = curr.next
        return self.head
    


        
        
        